options/gridcomponents.js

```diff
@@ -125,20 +125,22 @@ class DeleteButtonCellRenderer {
 
 }
 
-/**
- *  Cell editor for the input type. Shows a dropdown menu with the options.
- */
-class InputTypeEditor {
+class DropdownEditor {
+
+  /** @type {string} */
+  colId;
+  /** @type {string[]} */
+  options = [];
 
   static getOption(params, value, text) {
     return '<option value="' + value + '"' + (value == params.value ? ' selected' : '') + '>' + text + '</option>';
   }
 
   init(params) {
     this.gui = $('<select>' +
-                  InputTypeEditor.getOption(params, 0, browser.i18n.getMessage("inputType.text")) +
-                  InputTypeEditor.getOption(params, 1, browser.i18n.getMessage("inputType.wholeWords")) +
-                  InputTypeEditor.getOption(params, 2, browser.i18n.getMessage("inputType.regExp")) +
+                  this.options.reduce((accum, curOption, i) =>
+                    DropdownEditor.getOption(params, i, browser.i18n.getMessage(curOption)) + accum
```

Comment:
The dropdown elements appear in reverse order from specified in code. The concatenation should be `accum + DropdownEditor.getOption(...)` instead.